File: frontend/src/lib/api/generation.ts
import { apiClient } from "./client";

export interface GenerationRequest {
  topic: string;
  client_id: string;
  template_id?: string;
  tone?: string;
  keywords?: string[];
  services?: string[];
  custom_direction?: string;
  auto_distribute?: boolean;
  distribution_platforms?: string[];
}

export interface GenerationStatus {
  job_id: string;
  document_id: string;
  status: string;
  current_step: string | null;
  progress_percent: number;
  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;
}

export interface GenerationJob {
  id: string;
  document_id: string;
  status: string;
  current_step: string | null;
  progress_percent: number;
  steps: Record<string, unknown> | null;
  started_at: string | null;
  completed_at: string | null;
  tokens_used: number | null;
  api_cost: number | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export const generationApi = {
  generate: async (data: GenerationRequest): Promise<GenerationStatus> => {
    const response = await apiClient.post("/generation/generate", data);
    return response.data;
  },

  getJob: async (jobId: string): Promise<GenerationJob> => {
    const response = await apiClient.get(`/generation/jobs/${jobId}`);
    return response.data;
  },

  cancelJob: async (jobId: string): Promise<void> => {
    await apiClient.post(`/generation/jobs/${jobId}/cancel`);
  },

  retryJob: async (jobId: string): Promise<void> => {
    await apiClient.post(`/generation/jobs/${jobId}/retry`);
  },
};

File: backend/app/schemas/generation.py
"""Generation schemas."""

from typing import Optional, List, Dict, Any
from datetime import datetime

from pydantic import BaseModel


class GenerationRequest(BaseModel):
    """Content generation request schema."""

    topic: str
    client_id: str
    template_id: Optional[str] = None
    tone: Optional[str] = None
    keywords: Optional[List[str]] = None
    services: Optional[List[str]] = None
    custom_direction: Optional[str] = None
    auto_distribute: bool = False
    distribution_platforms: Optional[List[str]] = None


class GenerationStatus(BaseModel):
    """Generation status response schema."""

    job_id: str
    document_id: str
    status: str
    current_step: Optional[str] = None
    progress_percent: int
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    error_message: Optional[str] = None


class GenerationJobResponse(BaseModel):
    """Generation job response schema."""

    id: str
    document_id: str
    status: str
    current_step: Optional[str] = None
    progress_percent: int
    steps: Optional[Dict[str, Any]] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    tokens_used: Optional[int] = None
    api_cost: Optional[float] = None
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
